rosidl_parser/rosidl_parser/grammar.lark

```diff
@@ -0,0 +1,518 @@
+
+// (16)
+primary_expr: scoped_name
+  | literal
+  | "(" const_expr ")"
+
+// (17)
+literal.2: integer_literal
+  | floating_pt_literal
+  | fixed_pt_literal
+  | character_literal
+  | wide_character_literal
+  | boolean_literal
+  | string_literal
+  | wide_string_literal
+
+// (18)
+boolean_literal: boolean_literal_true
+  | boolean_literal_false
+// separate rules to identify the boolean literal
+boolean_literal_true: "TRUE"
+boolean_literal_false: "FALSE"
+
+// (19)
+positive_int_const: const_expr
+
+// (20)
+type_dcl: constr_type_dcl
+  | native_dcl
+  | typedef_dcl
+
+// (21), (216)
+type_spec: simple_type_spec
+  | template_type_spec
+
+// (22)
+simple_type_spec: base_type_spec
+  | scoped_name
+
+// (23)
+base_type_spec.2: integer_type
+  | floating_pt_type
+  | char_type
+  | wide_char_type
+  | boolean_type
+  | octet_type
+
+// (24)
+floating_pt_type.2: floating_pt_type_float
+  | floating_pt_type_double
+  | floating_pt_type_long_double
+// separate rules to identify the floating point type
+floating_pt_type_float: "float"
+floating_pt_type_double: "double"
+floating_pt_type_long_double: "long" "double"
+
+// (25)
+integer_type.2: signed_int
+  | unsigned_int
+
+// (26), (206)
+signed_int: signed_short_int
+  | signed_long_int
+  | signed_longlong_int
+  | signed_tiny_int
+
+// (27), (210)
+signed_short_int: "short"
+  | "int16"
+
+// (28), (211)
+signed_long_int: "long"
+  | "int32"
+
+// (29), (212)
+signed_longlong_int: "long" "long"
+  | "int64"
+
+// (30), (207)
+unsigned_int: unsigned_short_int
+  | unsigned_long_int
+  | unsigned_longlong_int
+  | unsigned_tiny_int
+
+// (31), (213)
+unsigned_short_int: "unsigned" "short"
+  | "uint16"
+
+// (32), (214)
+unsigned_long_int: "unsigned" "long"
+  | "uint32"
+
+// (33), (215)
+unsigned_longlong_int: "unsigned" "long" "long"
+  | "uint64"
+
+// (34)
+char_type.2: "char"
+
+// (35)
+wide_char_type.2: "wchar"
+
+// (36)
+boolean_type.2: "boolean"
+
+// (37)
+octet_type.2: "octet"
+
+// (38)
+template_type_spec.2: sequence_type
+  | string_type
+  | wide_string_type
+  | fixed_pt_type
+
+// (39)
+sequence_type: "sequence" "<" type_spec "," positive_int_const ">"
+  | "sequence" "<" type_spec ">"
+
+// (40)
+string_type.2: "string" "<" positive_int_const ">"
+  | "string"
+
+// (41)
+wide_string_type.2: "wstring" "<" positive_int_const ">"
+  | "wstring"
+
+// (42)
+fixed_pt_type: "fixed" "<" positive_int_const "," positive_int_const ">"
+
+// (43)
+fixed_pt_const_type.2: "fixed"
+
+// (44)
+constr_type_dcl: struct_dcl
+  | union_dcl
+  | enum_dcl
+
+// (45)
+struct_dcl: struct_def
+  | struct_forward_dcl
+
+// (46), 7.4.15.2
+struct_def: annotation_appl* "struct" IDENTIFIER "{" member+ "}"
+
+// (47), 7.4.15.2
+member: annotation_appl* type_spec declarators ";"
+
+// (48)
+struct_forward_dcl: "struct" IDENTIFIER
+
+// (49)
+union_dcl: union_def
+  | union_forward_dcl
+
+// (50)
+union_def: "union" IDENTIFIER "switch" "(" switch_type_spec ")" "{" switch_body "}"
```

Comment:
I assumed unions wouldn't be supported. Will the generators generate code for them?

Reply:
Correct assumption, no the pipeline doesn't support them.

I should check if I can remove those from the grammar instead.

Reply:
Roger. I think it can be in the grammar as long as something above warn's or errors when it sees a union.